### DISCUSSION TITLE
Making the malt node sockets match their blender counterparts

### DISCUSSION THREAD

**Discussion by Kolupsy** [bnpr/Malt#268] [Feedback & Feature Requests]
I personally think it would be nice if the socket colors could match the colors from builtin sockets
Its mostly the vec3 and vec4 that are currently too similar to be confusable. This is a table of socket colors used:
```
Float [0.63, 0.63, 0.63, 1.0]
Color [0.78, 0.78, 0.16, 1.0]
Integer [0.35, 0.55, 0.36, 1.0]
String [0.44, 0.7, 1.0, 1.0]
Boolean [0.8, 0.65, 0.84, 1.0]
Materal [0.92, 0.46, 0.51, 1.0]
Texture [0.62, 0.31, 0.64, 1.0]
Object [0.93, 0.62, 0.36, 1.0]
Image [0.39, 0.22, 0.39, 1.0]
Vector [0.39, 0.39, 0.78, 1.0]
Geometry [0.0, 0.84, 0.64, 1.0]
Shader [0.39, 0.78, 0.39, 1.0]
```

**Comment by pragma37** [comment]
I think it won't translate well. For example, Blender uses vec3 for Color and Vector, but Color can be vec4 too.

If you feel like testing it, you could try adding some defaults to the __TYPE_COLORS dictionary in MaltSockets.
https://github.com/bnpr/Malt/blob/103f4dd6b9838f974aff13ec39f2849c345d9b1e/BlenderMalt/MaltNodes/MaltSocket.py#L3

**Reply by Kolupsy** [reply]
In general colors are vec4 and yellow and vectors vec3 and purple. why the alpha is never exposed in shader nodes is a mystery to me too but yellow sockets are always vec4 afaik.
This is what it looks like with blenders color scheme. Looks a bit off though because the socket borders are gray and not black like in geometry nodes and the shader editor
![image](https://user-images.githubusercontent.com/88381426/157867213-b353128c-fcb5-4f6f-93a7-381819f26e40.png)
